turbinia/evidence.py

```diff
@@ -252,10 +256,10 @@ class ChromiumProfile(Evidence):
 
   def __init__(self, browser_type=None, output_format=None, *args, **kwargs):
     """Initialization for chromium profile evidence object."""
+    self.copyable = True
```

Comment:
yup, I'm stupid.
This should either be an optional parameter to Evidence.__init__() or a Setter() WDYT?

Reply:
Basically I want to raise when someone tries to subclass an Event which has set copyable=True and hasn't provided a source_path. So either:

```
class Evidence
def __init__(self,.....,copyable=False):
   self.copyable = copyable

  if self.copyable and not self.source_path:
    raise

class SubEvidence()
  def __init__(self, *args, **kwargs):
    super(B, self).__init__(copyable=True, *args, **kwargs)
```
or make self.copyable an @property (and I check I have a source_path in the setter)

Reply:
Yeah, I think I like the first version better (re-passing the var into the super class constructor).  I'm pretty sure there are other examples of that in evidence objects, or at least elsewhere in Turbinia.

Reply:
So it turns out this solution breaks decoding evidence objects, as we previously would call [Evidence.__init__() ](https://github.com/google/turbinia/blob/master/turbinia/evidence.py#L60) and then set the object's __dict__
This breaks now because of the check in __init__.
So I kind of went with a similar idea as [this](https://gist.github.com/ardenn/30f94f57876a70832a5c960fd4742d89#file-json_dict_to_obj-py) where I have code to properly pull all named parameters from the serialized dict, pass them to __init__(), and then update __dict__ with whatever left

Reply:
... which still poses problem. I do like:
```
class Evidence(object):
    @classmethod
    def from_dict(cls, dictionary):
      """Build an Evidence object from a serialized dict."""
      name = dictionary.pop('name', None)
      description = dictionary.pop('description', None)
      source = dictionary.pop('source', None)
      source_path = dictionary.pop('source_path', None)
      tags = dictionary.pop('tag', None)
      request_id = dictionary.pop('request_id', None)
      copyable = dictionary.pop('copyable', False)
      new_object = cls(
          name=name, description=description, source=source,
          source_path=source_path, tags=tags, request_id=request_id,
          copyable=copyable)
      new_object.__dict__.update(dictionary)
      return new_object
```

Problem is then than I have twice the "copyable=" part now, for example in:
```
  class PlasoFile(Evidence):
    def __init__(self, plaso_version=None, *args, **kwargs):
           super(PlasoFile, self).__init__(copyable=True, *args, **kwargs)
```
I was playing with the idea of checking whether 'copyable' is already in kwargs there.... But I also kind of wished we stopped using " *args, **kwargs" everywhere =)